cmake/Modules/FindMKL.cmake

```diff
@@ -212,7 +216,7 @@ MACRO(CHECK_ALL_LIBRARIES LIBRARIES OPENMP_TYPE OPENMP_LIBRARY _name _list _flag
         # Separately handling compiled TBB
         SET(_found_tbb TRUE)
       ELSE()
-        FIND_LIBRARY(${_prefix}_${_library}_LIBRARY NAMES ${_library})
+        FIND_LIBRARY(${_prefix}_${_library}_LIBRARY NAMES ${_library} ${_library}_dll)
```

Comment:
The DLLs with the suffix `_dll` are used for shared libraries. We use static libraries here.

Reply:
use BUILD_SHARED_LIBS  to check build type